hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/table/HoodieSparkTable.java

```diff
@@ -99,9 +100,16 @@ protected Option<HoodieTableMetadataWriter> getMetadataWriter(
       // Create the metadata table writer. First time after the upgrade this creation might trigger
       // metadata table bootstrapping. Bootstrapping process could fail and checking the table
       // existence after the creation is needed.
-      HoodieTableMetadataWriter metadataWriter = SparkHoodieBackedTableMetadataWriter.create(
-          context.getHadoopConf().get(), config, failedWritesCleaningPolicy, context,
-          Option.of(triggeringInstantTimestamp));
+      HoodieTableMetadataWriter metadataWriter;
+      if (config.shouldEnableBootstrapMetadataSync()) {
+        metadataWriter = SparkHoodieBackedMetadataSyncMetadataWriter.create(
```

Comment:
why do we need this change?
we are directly instantiating `SparkHoodieBackedMetadataSyncMetadataWriter` within `HoodieMetadataSync` right. 
So, why we need these changes.